### ISSUE TITLE
AssemblyInfoFile - Path cannot be the empty string or all whitespace

### ISSUE THREAD

**Issue by DaveHogan** [fsharp/FAKE#996]
I think there is possibly a bug in `AssemblyInfoFile.fs` when the outputFileName does not contain any directory path. Either that, or I'm doing something whacky.

The scenario I have is, the generated file resides on the root of the application so in my build script I have:

`CreateCSharpAssemblyInfo "CommonAssemblyInfo.cs"` (without any directory information)

I believe this function here: 

```FSharp
let private writeToFile outputFileName (lines : seq<string>) = 
    let fi = fileInfo outputFileName
    if fi.Exists then fi.Delete()
    System.IO.Directory.CreateDirectory(System.IO.Path.GetDirectoryName(outputFileName)) |> ignore
    use writer = new System.IO.StreamWriter(outputFileName, false, System.Text.Encoding.UTF8)
    lines |> Seq.iter writer.WriteLine
    tracefn "Created AssemblyInfo file \"%s\"." outputFileName
```

Needs a conditional check like so. 
```FSharp
let private writeToFile outputFileName (lines : seq<string>) = 
    let fi = fileInfo outputFileName
    if fi.Exists then fi.Delete()
    let dirName = System.IO.Path.GetDirectoryName(outputFileName)
    if not string.IsNullOrEmpty(dirName) then
      System.IO.Directory.CreateDirectory(dirName) |> ignore
    use writer = new System.IO.StreamWriter(outputFileName, false, System.Text.Encoding.UTF8)
    lines |> Seq.iter writer.WriteLine
    tracefn "Created AssemblyInfo file \"%s\"." outputFileName
```

I'm not yet tested it and unlikely to do so until next week so I thought I'd raise as an issue in the meantime.

The actual exception from the build script is 

```
Running build failed.
Error:
System.ArgumentException: Path cannot be the empty string or all whitespace.
   at System.IO.Directory.CreateDirectory(String path)
   at Fake.AssemblyInfoFile.writeToFile(String outputFileName, IEnumerable`1 lines) in C:\code\fake\src\app\FakeLib\AssemblyInfoFile.fs:line 104
   at Fake.AssemblyInfoFile.CreateCSharpAssemblyInfoWithConfig(String outputFileName, IEnumerable`1 attributes, AssemblyInfoFileConfig config) in C:\code\fake\s
rc\app\FakeLib\AssemblyInfoFile.fs:line 142
   at FSI_0001.Build.clo@51-2.Invoke(Unit _arg2) in i:\Repositories\LizEarleBeautyCoLtd\amo_admin\build.fsx:line 52
   at Fake.TargetHelper.runSingleTarget(TargetTemplate`1 target) in C:\code\fake\src\app\FakeLib\TargetHelper.fs:line 483
```

This was discovered using v.4.3.4 however looking at the code, it still appears to be a problem?

**Comment by forki** [comment]
please send a pull request with the fix

**Comment by le-dhogan** [comment]
:+1: Will do over the weekend

**Comment by le-dhogan** [comment]
@forki - this can be closed now. Fixed in 4.9.1
